src/core/Song.cpp

```diff
@@ -667,17 +667,11 @@ void Song::stop()
 					m_playPos[m_playMode].setTicks(tl->savedPos().getTicks());
 					setToTime(tl->savedPos());
 
-					if( gui && gui->songEditor() &&
-							( tl->autoScroll() == TimeLineWidget::AutoScrollEnabled ) )
-					{
-						gui->songEditor()->m_editor->updatePosition( MidiTime(tl->savedPos().getTicks() ) );
-					}
 					tl->savePos( -1 );
 				}
 				break;
 
 			case TimeLineWidget::KeepStopPosition:
-			default:
```

Comment:
Generally speaking it should be sane for enums to have a default fall-through to prevents compilation errors if the enum is changed down the road.  I'm not sure why this was removed.  Did your code editor do this automatically?

Reply:
Hi @tresf,

thanks for having a look at the code! I have removed the `default` case because QtCreator highlighted it with a warning that there is a `default` case although all cases are handled.

I currently cannot see how a `default` case that does nothing should help in this place. If you add a new enumeration value then the effect of the current code will be the same like with the `default` case. Nothing happens if the new value is set somewhere. If you rename or remove a value from the enumeration then the compiler will tell you everywhere where it was used.

However, I am not too invested in this. If it's wanted I can put the `default` case back.

Reply:
> I have removed the default case because QtCreator highlighted it with a warning that there is a default case although all cases are handled.

Interesting.  Thanks.  I had a feeling this was IDE driven.  More of a query related to the behavior of the IDEs than a criticism of the decision since we're bound to see this happen more.  This enum is likely to stay fully cased due to it's design, but the `default` fall-through is often preferred for frequently changing enums.